### PR TITLE
Exposed the serializable-model when generating client models.

### PR DESCRIPTION
**Pull request by denvitaharen**
Many thanks for submitting your Pull Request :heart:!

Issue: https://github.com/quarkiverse/quarkus-openapi-generator/issues/869

Exposed the possibility to generate models that implement the `java.io.Serilizable`-interface. I changed the property from the proposed `serializableModel` to `serializable-model` since I thought it harmonized better with the naming of the other configurations.

I wrote an new test module for this, maybe it was a bit over engineered, but I didn't find any existing suitable test to put this in, so I did a new module instead.

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [x] Pull Request contains link to the issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by denvitaharen** [comment]
My bad, I had set IntelliJ to optimize imports and reformat code, which changed everything the `verify` had made. Rerunned it now and made sured IntelliJ didn't change anything.

File: client/integration-tests/serializable-model/src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+quarkus.openapi-generator.codegen.spec.quarkus_serializable_openapi_yaml.serializable-model=true
+quarkus.openapi-generator.codegen.spec.quarkus_serializable_openapi_yaml.base-package=io.quarkiverse.serializable
+quarkus.openapi-generator.codegen.spec.quarkus_non_serializable_openapi_yaml.base-package=io.quarkiverse.non.serializable
```
Comment (ricardozanini) [inline client/integration-tests/serializable-model/src/main/resources/application.properties:3]: Please use `org.acme` instead. `io.quarkiverse` can lead to confusion.

Comment (denvitaharen) [reply client/integration-tests/serializable-model/src/main/resources/application.properties]: Done.

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java [hunk 2]
```diff
@@ -1,23 +1,7 @@
...
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.INPUT_BASE_DIR;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.MODEL_NAME_PREFIX;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.MODEL_NAME_SUFFIX;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.REMOVE_OPERATION_ID_PREFIX;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.REMOVE_OPERATION_ID_PREFIX_COUNT;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.REMOVE_OPERATION_ID_PREFIX_DELIMITER;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.TEMPLATE_BASE_DIR;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.VALIDATE_SPEC;
+import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.*;
+import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.ConfigName.*;
```
[the remark this answers is not in the document]
Comment (denvitaharen) [reply client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java]: Yes. I reverted the change.

File: client/integration-tests/pom.xml [hunk 3]
```diff
@@ -44,6 +44,7 @@
     <module>config-key</module>
     <module>github</module>
     <module>without-oidc</module>
+      <module>serializable-model</module>
```
Comment (hbelmiro) [inline client/integration-tests/pom.xml:47]: Please fix the indentation.

```suggestion
    <module>serializable-model</module>
```

File: client/integration-tests/serializable-model/src/test/java/io/quarkiverse/openapi/generator/it/QuarkusSerializableModelRestEasyReactiveTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.io.Serializable;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+@Tag("resteasy-reactive")
```
Comment (hbelmiro) [inline client/integration-tests/serializable-model/src/test/java/io/quarkiverse/openapi/generator/it/QuarkusSerializableModelRestEasyReactiveTest.java:13]: You only need one file for each RESTEasy implementation when they have different tests, which is not the case.
So, you can keep one single file and remove `@Tag`. The CI will be in charge of testing both implementations using profiles.

For more info: https://github.com/quarkiverse/quarkus-openapi-generator/blob/main/CONTRIBUTING.md#resteasy-reactive-and-resteasy-classic

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapper.java [hunk 5]
```diff
@@ -1,8 +1,6 @@
 package io.quarkiverse.openapi.generator.deployment.wrapper;
 
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.getSanitizedFileName;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.resolveApiPackage;
-import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.resolveModelPackage;
+import static io.quarkiverse.openapi.generator.deployment.CodegenConfig.*;
```
Comment (hbelmiro) [inline client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapper.java:3]: Please don't use star imports.

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapper.java [hunk 6]
```diff
@@ -11,11 +9,7 @@
 
 import java.io.File;
 import java.nio.file.Path;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (hbelmiro) [inline client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapper.java:12]: Please don't use star imports.

**Review by hbelmiro — CHANGES_REQUESTED** [review]
Thank you @denvitaharen.
Looks great in general. I just left a few minor comments.

**Comment by denvitaharen** [comment]
Thanks for the comments @hbelmiro! I have updated the PR based on your feedback.

Out off curiosity, why isn't star imports ok? I don't have any personal preferences, I just use what's default in my IDE and have never reflected or hade any problems with star imports :)

**Review by hbelmiro — APPROVED** [review]

**Comment by hbelmiro** [comment]
> Thanks for the comments @hbelmiro! I have updated the PR based on your feedback.
> 
> Out of curiosity, why isn't star imports ok? I don't have any personal preferences, I just use what's default in my IDE and have never reflected or hade any problems with star imports :)

@denvitaharen star imports don't bring any benefits, but some disadvantages:
* Reduce readability since it's unclear which package a class comes from
* Reduce code maintainability: future changes in dependencies or new class additions to imported packages might introduce conflicts or ambiguities

**Pull request merged**